packages/react/src/create-strict-api/index.ts

```diff
@@ -59,7 +59,9 @@ interface CompiledAPI<TSchema> {
       StrictCSSProperties & PseudosDeclarations & EnforceSchema<TSchema>
     >
   >(
-    styles: TStyles
+    // NOTE: This should be 1:1 with the `TStyles extends …` above as we want it to strictly satisfy this type, not extend it.
+    // The "extends" functionality is to infer and build the return type, this is to enforce the input type
+    styles: Record<string, StrictCSSProperties & PseudosDeclarations & EnforceSchema<TSchema>>
```

Comment:
Nit: We could potentially extract this type (and other similar types in this file, like the parameter for `css`) into a separate variable to minimise repetition, e.g. `type CssMapStyles = Record<string, ...>;`

Reply:
Yeah, that's valid, doing into:
```ts
type CSSStyles<TSchema extends CompiledSchema> = StrictCSSProperties &
  PseudosDeclarations &
  EnforceSchema<TSchema>;
type CSSMapStyles<TSchema extends CompiledSchema> = Record<string, CSSStyles<TSchema>>;
```

Reply:
Done!